### PR TITLE
Minor clean up of difftool patch

### PR DESCRIPTION
**Pull request by rtyler**
See original PR: #422

### CODE REVIEW

File: manifests/job.pp [hunk 1]
```diff
@@ -31,6 +31,7 @@
   $jobname  = $title,
   $enabled  = 1,
   $ensure   = 'present',
+  $difftool = 'diff -b -q',
```
Comment (jhoblitt) [inline manifests/job.pp]: The param should have some sort of validation, probably `validate_string()`.

File: manifests/job/present.pp [hunk 2]
```diff
@@ -17,6 +17,7 @@
   $config,
   $jobname  = $title,
   $enabled  = 1,
+  $difftool = 'diff -b -q',
```
Comment (jhoblitt) [inline manifests/job/present.pp]: Ditto

File: manifests/job/present.pp [hunk 3]
```diff
@@ -78,7 +79,7 @@
   exec { "jenkins update-job ${jobname}":
     command => "${cat_config} | ${update_job}",
     onlyif  => "test -e ${config_path}",
-    unless  => "diff -b -q ${config_path} ${tmp_config_path}",
+    unless  => "${difftool} ${config_path} ${tmp_config_path}",
```
Comment (jhoblitt) [inline manifests/job/present.pp]: A doc note suggesting fully qualifying the path to the diffing utility might not be a bad idea.  I can imagine issues being opened over a utility installed in `/usr/local/bin`/etc. not working.

**Comment by jhoblitt** [comment]
This gives me a lot less heartburn than adding a pip dependency. I do have some concern about replicating this semantic in a provider but I guess we could punt and remove it as part of a major API bump when the time comes.

**Comment by rtyler** [comment]
Thanks for the review @jhoblitt, I'll take a crack at cleaning this up before the end of the day

**Comment by jhoblitt** [comment]
:+1:

**Pull request merged**